app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -71,6 +72,22 @@ export default function ExploreDisplayTabs({
   const [isCellSelecting, setIsCellSelecting] = useState(false)
   // a plotly points_selected event
   const [currentPointsSelected, setCurrentPointsSelected] = useState(null)
+
+  // Differential expression settings
+  // TODO (SCP-4321): Try encapsulating these in a DE-specific component,
+  // to simplify this high-level ExploreDisplayTabs component
+  const [showDeGroupPicker, setShowDeGroupPicker] = useState(false)
+  const [deGenes, setDeGenes] = useState(null)
```

Comment:
Ideally, the `genes` and `setDeGenes` state should live inside the DifferentialExpression component, since it's the one fetching them, and they're DE-specific info.  A `showFullPanelDe` boolean will make the logic below easier to read, and it will give the DE component the ability to switch/reload/clear the gene array without hiding itself.

Reply:
Thanks for chatting with me about this just now; I think 02bd684 addresses your main concern.  I suspect further simplifications will tightly couple with UX refinements suggested at this week's demo, and those seem best scoped for the next PR that matures the DE frontend.